### PR TITLE
Removes  dependency on rustc_serialize (#5988)

### PR DESCRIPTION
**Pull request by 0x7CFE**
Affects #5988.

- `util/network` was migrated to serde 1.0
- ethcore bench was fixed and adapted to use `rustc_hex`

Looks like these were the last dependencies that used `rustc_serialize`. 

Default test suite passes: 
```
test result: ok. 84 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

Fixed bench shows the following stats:
```
running 4 tests
test bn_128_mul     ... bench:     619,273 ns/iter (+/- 12,419)
test bn_128_pairing ... bench:   6,367,055 ns/iter (+/- 10,457)
test ecrecover      ... bench:     117,225 ns/iter (+/- 7,253)
test sha256         ... bench:       1,227 ns/iter (+/- 64)

test result: ok. 0 passed; 0 failed; 0 ignored; 4 measured; 0 filtered out
```
Just in case, here are my CPU specs:
```
lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                12
On-line CPU(s) list:   0-11
Thread(s) per core:    2
Core(s) per socket:    6
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 63
Model name:            Intel(R) Core(TM) i7-5930K CPU @ 3.50GHz
Stepping:              2
CPU MHz:               1281.738
CPU max MHz:           4000,0000
CPU min MHz:           1200,0000
BogoMIPS:              7012.16
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              15360K
NUMA node0 CPU(s):     0-11
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm epb tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm xsaveopt cqm_llc cqm_occup_llc dtherm ida arat pln pts
```

### CODE REVIEW

**Comment by NikVolf** [comment]
I don't think benchmarks data is relevant here, only code executed `Bencher::iter` is measured

**Comment by rphmeier** [comment]
Doesn't the whisper crate currently use 0.9?

**Comment by debris** [comment]
@rphmeier no, only `1.0` is used now

File: Cargo.lock [hunk 1]
```diff
@@ -811,6 +811,7 @@ dependencies = [
  "rust-crypto 0.2.36 (registry+https://github.com/rust-lang/crates.io-index)",
  "rustc-hex 1.0.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "rustc-serialize 0.3.24 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (debris) [inline Cargo.lock]: I just noticed, you most forgot to remove `rust-serialize` from `ethcore/Cargo.toml` and and `util/network/Cargo.toml`

Comment (0x7CFE) [reply Cargo.lock]: My bad, looks like I searched using underscore instead of dash.

**Pull request merged**